Tests/UITests/TestAppUITests/SpeziOnboardingTests.swift

```diff
@@ -72,7 +72,15 @@ final class OnboardingTests: XCTestCase { // swiftlint:disable:this type_body_le
         
         XCTAssert(app.buttons["Next"].waitForExistence(timeout: 2))
         app.buttons["Next"].tap()
-        
+
+        XCTAssert(app.staticTexts["Leland"].waitForExistence(timeout: 2))
+        XCTAssert(app.buttons["Next"].waitForExistence(timeout: 2))
+        app.buttons["Next"].tap()
+
+        XCTAssert(app.staticTexts["Stanford"].waitForExistence(timeout: 2))
+        XCTAssert(app.buttons["Next"].waitForExistence(timeout: 2))
+        app.buttons["Next"].tap()
+
```

Comment:
We assert that these views are inserted in the right order. But are we asserting that their identifiers are set as expected within those tests?

Reply:
At the moment we are not, we cannot test the internal state from a UI Test so the only possibility here would be to assert the value of a `TextView` which would have to access the `id` from its wrapping `OnboardingIdentifiableViewModifier` which is all too much of an effort for such a basic assertion imo.

Reply:
I would highly recommend that we add a test for that. It's very implicit that this works currently (we rely on the Identifiable extension of `ModifiedContent`) and it might easily break without someone noticing with future changes.

I agree, that it is pretty clumsy to do this using a UI test. I would propose the following unit test that would verify this behavior and prevent future regressions. Just add it as part of the unit tests of the swift package.
I would argue that we can make the `firstOnboardingStepIdentifier` property internal (currently it is private) to support this unit test.
```swift
    @MainActor
    func testOnboardingIdentifierModifier() throws {
        let stack = OnboardingStack {
            Text("Hello World")
                .onboardingIdentifier("Custom Identifier")
        }

        let identifier = try XCTUnwrap(stack.onboardingNavigationPath.firstOnboardingStepIdentifier)

        var hasher = Hasher()
        hasher.combine("Custom Identifier")
        let final = hasher.finalize()
        XCTAssertEqual(identifier.identifierHash, final)
    }
```